### PR TITLE
Synchronous invites

### PR DESCRIPTION
**Pull request by neilalexander**
This PR removes invite queues from the federation sender and instead wires up the APIs so that they are sent synchronously. The room server will send a federation request if it is required.

### CODE REVIEW

File: roomserver/api/perform.go [hunk 1]
```diff
@@ -105,8 +105,6 @@ type PerformInviteRequest struct {
 }
 
 type PerformInviteResponse struct {
-	// If non-nil, the invite request failed. Contains more information why it failed.
-	Error *PerformError
```
Comment (Kegsay) [inline roomserver/api/perform.go:109]: Why was this removed? The lack of a dedicated type means we can't return appropriate error codes to the client.

Comment (neilalexander) [reply roomserver/api/perform.go:109]: I did wonder about the API shape of that. I can put something back in.
